attalos/imgtxt_algorithms/approaches/negsampling.py

```diff
@@ -71,55 +72,58 @@ def meanlogsig(predictions, truth):
     def __init__(self, wv_model, datasets, **kwargs):
         self.wv_model = wv_model
         self.one_hot = OneHot(datasets, valid_vocab=wv_model.vocab)
+        word_counts = NegativeSampler.get_wordcount_from_datasets(datasets, self.one_hot)
+        self.negsampler = NegativeSampler(word_counts)
         train_dataset = datasets[0] # train_dataset should always be first in datasets
         self.w = construct_W(wv_model, self.one_hot.get_key_ordering()).T
 
         self.learning_rate = kwargs.get("learning_rate", 0.0001)
         self.optim_words = kwargs.get("optim_words", True)
         self.hidden_units = kwargs.get("hidden_units", "200,200")
         self.hidden_units = [int(x) for x in self.hidden_units.split(",")]
-
         self.model_info = self._construct_model_info(
             input_size = train_dataset.img_feat_size,
             output_size = self.one_hot.vocab_size,
+            hidden_units=self.hidden_units,
             learning_rate = self.learning_rate,
             optim_words = self.optim_words,
             wv_arr = self.w
         )
         self.test_one_hot = None
         self.test_w = None
         super(NegSamplingModel, self).__init__()
-
-    def _get_ids(self, pBatch, distribution=None, numSamps=[5,10]):
-        nBatch = 1.0 - pBatch
-        vpia = []; vnia = [];
-        for i,unisamp in enumerate(pBatch):
-            if distribution is None:
-                p=1.0*unisamp/sum(unisamp)
-            else:
-                p = distribution
-            vpi = np.random.choice( range(len(unisamp)) , size=numSamps[0],  p=p)
-            vpia += [vpi]
-        for i,unisamp in enumerate(nBatch):
-            if distribution is None:
-                p = 1.0 * unisamp / sum(unisamp)
-            else:
-                p = distribution
-            vni = np.random.choice( range(len(unisamp)) , size=numSamps[1], p=p)
-            vnia += [vni]
-        vpia = np.array(vpia)
-        vnia = np.array(vnia)
-        return vpia, vnia
+    
+    def _get_ids(self, tag_ids, numSamps=[5, 10], uniform_sampling=False):
+        """
+        Takes a batch worth of text tags and returns positive/negative ids
+        """
+        pos_word_ids = np.ones((len(tag_ids), numSamps[0]))*-1
```

Comment:
Is there a reason why pos_word_ids is initialized with -1's and neg_word_ids seems like the default is 0?